cpp/arcticdb/storage/test/test_embedded.cpp

```diff
@@ -16,182 +16,185 @@
 #include <filesystem>
 #include <stdexcept>
 #include <arcticdb/entity/atom_key.hpp>
-#include <arcticdb/util/buffer.hpp>
 #include <arcticdb/entity/types.hpp>
 #include <arcticdb/util/test/test_utils.hpp>
 #include <arcticdb/util/random.h>
 #include <arcticdb/stream/row_builder.hpp>
-#include <arcticdb/stream/aggregator.hpp>
+
+namespace test_embedded {
```

Comment:
I know it's not your naming, but I'm a bit at a loss to explain what 'test_embedded' actually tests, or what a BackendGenerator is (a thing that makes storages, apparently). Also, I think that anonymous namespaces are probably fine for cpp files in test code, we don't need to give them unique namespace names.

Reply:
It looks like this is the one that tests local storages, so the generator could be a LocalStorageGenerator, and in the tet below we have a RemoteStorageGenerator that make the other kinds

Reply:
Thanks, I've tidied this up quite a bit and put the Generator somewhere it can be shared across the two test files.